ProfileService.js

```diff
@@ -68,10 +64,17 @@ export class ProfileService {
   async createAgent(
     {url = this.config.urls.profileAgents, account, profile, token} = {}) {
     try {
-      const response = await this._axios.post(url, {account, profile, token});
+      if(url.startsWith('/')) {
+        url = url.substring(1);
+      }
```

Comment:
String manipulation for urls passed into the function should be a red flag. Can we configure a single `http-client` instance and make that a property of the class? This would replace what was previously named `_axios`.

Reply:
+1, agree with Ganesh, that this isn't the right approach.

Instead, we should use the URL class (and its handling of base URLs, paths and slashes).
So, the constructor for the [URL class](https://nodejs.org/docs/latest-v14.x/api/url.html#url_new_url_input_base) is:
`new URL(input, base)` -- exactly what we're trying to do here. So the code should be more like:

```js
      const response = await httpClient.post(
        new URL(url, this.baseURL).toString(), {
          json: {account, didMethod, didOptions}
        });
```

Reply:
What Dmitri proposed is a solution that should be used if no other avenues present themselves. It is not ideal to parse a url and serialize it to a string for every request. If we can get back to a simple string concatenation, that would be ideal.

Reply:
@gannan08 the http-client lib exports an instance of http client https://github.com/digitalbazaar/http-client/blob/master/main.js#L16. It doesn't look like there's an easy way to make a new instance so I am not sure how to do this?